packages/eslint-config-nimble/lint.js

```diff
@@ -0,0 +1,22 @@
+import { defineConfig } from 'eslint/config';
+import { javascriptConfig, importNodeEsmConfig } from '@ni/eslint-config-javascript';
+
+export const lintNimbleConfig = defineConfig([
+    {
+        files: [
+            '**/eslint.config.cjs',
+            '**/eslint.config.cts',
+            '**/eslint.config.js',
+            '**/eslint.config.mjs',
+            '**/eslint.config.mts',
+            '**/eslint.config.ts'
+        ],
+        extends: [javascriptConfig, importNodeEsmConfig],
+        // Ignoring imports in each specific project eslint config
+        // so each package.json does not need explicit `eslint`
+        // and `eslint-config-nimble` dependencies.
+        rules: {
+            'import/no-extraneous-dependencies': 'off'
+        }
```

Comment:
I'm on the fence about this rule. What's its doing is ignoring the `package.json` dependency check specifically just for `eslint.config.*` files. Otherwise each package with eslint would need specific dependencies on `eslint` and the configs they depend on, i.e. `@ni/eslint-config-javascript` because each file imports from them.

As I write this I'm more convinced we should just do the dependency definitions correctly since packages have various purpose specific dependencies that should be tracked (react specific / storybook specific, etc). I'll keep it as a personal list background task follow-up PR unless someone feels strongly to do it in an immediate follow-on.